cfme/physical/physical_switch.py

```diff
@@ -0,0 +1,238 @@
+        """Wait for PhysicalSwitch to come to desired state. This function waits just the needed amount of
+           time thanks to wait_for.
+
+        Args:
+            desired_state (str): 'on' or 'off'
+            target (str): The name of the method that most be used to compare with the desired_state
+            view (object): The view that most be refreshed to verify if the value was changed
+            provider (object): 'LenovoProvider'
+            timeout (int): Specify amount of time (in seconds) to wait until TimedOutError is raised
+            delay (int): Specify amount of time (in seconds) to repeat each time.
+        """
+
+        def _is_state_changed():
+            self.refresh(provider, handle_alert=True)
+            return desired_state == getattr(self, target)()
+
+        wait_for(_is_state_changed, fail_func=view.browser.refresh, num_sec=timeout, delay=delay)
+
+    @property
+    def exists(self):
+        """Checks if the physical_switch exists in the UI.
+
+        Returns: :py:class:`bool`
+        """
+        view = navigate_to(self.parent, "All")
+        try:
+            view.entities.get_entity(name=self.name, surf_pages=True)
+        except ItemNotFound:
+            return False
+        else:
+            return True
+
+    @cached_property
+    def get_db_id(self):
+        if self.db_id is None:
+            self.db_id = self.appliance.physical_switch_id(self.name)
+            return self.db_id
+        else:
+            return self.db_id
+
+    def wait_to_appear(self):
+        """Waits for the Switch to appear in the UI."""
+        view = navigate_to(self.parent, "All")
+        logger.info("Waiting for the Switch to appear...")
+        wait_for(
+            lambda: self.exists,
+            message="Wait for the Switch to appear",
+            num_sec=1000,
+            fail_func=view.browser.refresh
+        )
+
+    def wait_for_delete(self):
+        """Waits for the Switch to remove from the UI."""
+        view = navigate_to(self.parent, "All")
+        logger.info("Waiting for the Switch to delete...")
+        wait_for(
+            lambda: not self.exists,
+            message="Wait for the Switch to disappear",
+            num_sec=500,
+            fail_func=view.browser.refresh
+        )
+
+
+@attr.s
+class PhysicalSwitchCollection(BaseCollection):
+    """Collection object for the :py:class:`cfme.infrastructure.host.PhysicalSwitch`."""
+
+    ENTITY = PhysicalSwitch
+
+    def select_entity_rows(self, physical_switches):
+        """ Select all physical Switch objects """
+        physical_switches = list(physical_switches)
+        checked_physical_switches = list()
+        view = navigate_to(self, 'All')
+
+        for physical_switch in physical_switches:
+            view.entities.get_entity(name=physical_switch.name, surf_pages=True).check()
+            checked_physical_switches.append(physical_switch)
+        return view
+
+    def all(self, provider):
```

Comment:
**REQUIRED**
We shouldn't provide any argument to `all()`. `filter` should be used to filter out returning values.

Reply:
Done.